text/0016-named-tracers.md

```diff
@@ -12,43 +12,53 @@ _Creating Tracers and Meters using a factory mechanism and naming those Tracers
+### Faulty or expensive instrumentation
 
-* If a library hasn't implemented [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md) correctly or those conventions change over time, it's currently hard to interpret and sanitize these data selectively. The produced Spans or Metrics cannot be associated with those instrumentation libraries later.
+For an operator of an application using OpenTelemetry, there is currently no way to influence the amount of data produced by reporting libraries. Reporting libraries can easily "spam" backend systems, deliver bogus data, or -- in the worst case -- crash or slow down applications. These problems might even occur suddenly in production environments because of external factors such as increasing load or unexpected input data.
+
+### Reporting library identification
+
+If a reporting library hasn't implemented [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md) correctly or those conventions change over time, it's currently hard to interpret and sanitize data produced by it selectively. The produced Spans or Metrics cannot later be associated with the library which reported them, either in the processing pipeline or the backend.
+
+### Disable instrumentation of pre-instrumented libraries
+
+It is the eventual goal of OpenTelemetry that library vendors implement the OpenTelemetry API, obviating the need to auto-instrument their library. An operator should be able to disable the telemetry that is built into some database driver or other library and provide their own integration if the built-in telemetry is lacking in some way. This should be possible even if the developer of that database driver has not provided a configuration to disable telemetry.
+
+## Solution
 
 This proposal attempts to solve the stated problems by introducing the concept of:
- * _Named Tracers and Meters_ identified via a **name** (e.g. _"io.opentelemetry.contrib.mongodb"_) and a **version** (e.g._"semver:1.0.0"_) which is associated with the Tracer / Meter and the Spans / Metrics it produces.
- * A `TracerFactory` / `MeterFactory` as the only means of creating a Tracer or Meter.
+ * _Named Tracers and Meters_ which are associated with the **name** (e.g. _"io.opentelemetry.contrib.mongodb"_) and **version** (e.g._"semver:1.0.0"_) of the library which acquired them.
+ * A `TracerRegistry` / `MeterRegistry` as the only means of acquiring a Tracer or Meter.
```

Comment:
Factory or Provider are by far much more accurate terms, because they are active, where as registry is passive.

 What is the justification for this change?

Reply:
I agree that registry is too passive. I can change it to provider. The justification was that a lot of people felt that "Factory" was too oop or java specific.

Reply:
> a lot of people felt that "Factory" was too oop or java specific

Factory is a design pattern that has nothing to do with OOP or Java. Even if it was, I don't see how "this feels like X" qualifies as an argument in a debate.

Reply:
I didn't say it was my argument, just that people felt that way. This was the original issue where the whole "language agnostic working group" stemmed from.

Reply:
Did that group also have an issue with "API is interface and SDK is implementation"? Because that also "feels like OOP" (because it is). Sorry for getting snarky, but I find that kind of reasoning very unprofessional - if people have arguments, let them state it in writing, as arguments. I've seen this factory issue bubble up several times, always with the same explanation "people feel that...", and never anyone stepping up, creating a ticket, and laying out convincing arguments for a different terminology. For all I know, I might agree with those arguments, but I can't even begin with  "this feels like ..." justification.

Reply:
I take issue with "Factory" because it implies something is being built. Nothing about this API demands that the SDK build something new when the factory is called. I believe "Provider" is the best term we have and it is the most literally accurate given what the current specification says.

Reply:
changed to provider